src/client/canvas/svg/arc.tsx

```diff
@@ -0,0 +1,67 @@
+import { observer } from 'mobx-react'
+import * as React from 'react'
+import { Component } from 'react'
+
+import { ArcGeometry } from '../geometry/arc_geometry'
+import { Shape } from '../object/shape'
+
+import { appearance } from './svg'
+
+@observer
+export class Arc extends Component<{model: Shape<ArcGeometry>}> {
+  // TODO draw arcs
+  public render() {
+    const m = this.props.model
+    const g = m.geometry
+    // g.anticlockwise
+    // g.startAngle
+    // g.endAngle
+    // g.origin
+    // g.radius
+    // rx ry x-axis-rotation large-arc-flag sweep-flag x y
+    return <path d={this.arc(g.origin,
+                             g.radius,
+                             g.startAngle,
+                             g.endAngle,
+                             g.anticlockwise)} {...appearance(m.appearance)} />
+  }
+
+
+  private arc(origin: {x: number, y: number},
+              r: number,
+              a0: number,
+              a1: number,
+              ccw: boolean): string {
+
+    // Is this arc empty?
+    if (r === 0) {
+      return ''
+    }
+    if (r < 0) {
+      throw new Error(`Negative radius: ${r}`)
+    }
+
+    // Work out our start and end points on the circle
+    const p0 = {
+      x: origin.x + r * Math.cos(a0),
+      y: origin.y + r * Math.sin(a0),
+    }
+
+    const p1 = {
+      x: origin.x + r * Math.cos(a1),
+      y: origin.y + r * Math.sin(a1),
+    }
+
+    // Our cw value must be a 0/1 not true/false
+    const cw = +!ccw
+    const da = ccw ? a0 - a1 : a1 - a0
+
+    // Move to (x0,y0).
+    let output = `M${p0.x} ${p0.y}`
+
+    // Draw the arc
+    output += `A${r} ${r} 0 ${+(da >= Math.PI)} ${cw} ${p1.x} ${p1.y}`
```

Comment:
Some of these render functions are pretty complicated, can we add some snapshot tests? They are pretty simple because you don't have to write the expected value manually. Here is one for circle:

```ts
import * as React from 'react'
import { BasicAppearance } from '../../appearance/basic_appearance'
import { CircleGeometry } from '../../geometry/circle_geometry'
import { Shape } from '../../object/shape'
import { Circle } from '../circle'

describe('CircleGeometry', () => {
  it('renders', () => {
    const model = Shape.of(CircleGeometry.of({
      radius: 2,
    }), BasicAppearance.of())
    expect(<Circle model={model}/>).toMatchSnapshot()
  })
})
```

Reply:
I'm not sure snapshots are the best way to make tests for these. It is valid to draw these in a different way which won't match the snapshot. E.g. if I changed the way I drew arcs to have a clippled circle for example, it wouldn't match the snapshot but it would be the same visually.

Reply:
The snapshots test serve to ensure that any future changes to the output are reviewed and intentional. If you wanted to change the way your drew arcs, you would just update the snapshots. Its mainly about allowing non-output changing refactors such that you are confident that the rendered output did not change from whatever it was before.

The type of visual regression testing you're proposing would be something achievable with something like [Percy](https://percy.io/), but in the mean time I think this is the next best thing.